### PR TITLE
Conversion functions from degrees to radians and vice-versa.

### LINKED ISSUE #837: Elemental conversion functions from degrees to radians and vice versa.
### Motivation

Would it be interesting to include in the standard library conversion functions between degrees and radians such as in Numpy?

### Prior Art

https://numpy.org/doc/stable/reference/generated/numpy.deg2rad.html

https://numpy.org/doc/stable/reference/generated/numpy.rad2deg.html

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by MilanSkocic**
As discussed in #837, I implemented: 
- [x] conversion functions: `deg2rad` and `rad2deg` in `stdlib_math`
- [x] documentation update
- [x] examples
- [x] tests 

Resolves #837

### CODE REVIEW

**Review by perazz — APPROVED** [review]
Thank you @MilanSkocic. LGTM with the edits and I think it is a useful addition. I would wait for an additional opinion from @jalvesz and @jvdp1.

File: doc/specs/stdlib_math.md [hunk 1]
```diff
@@ -382,6 +382,66 @@ Notes: Although the angle of the complex number `0` is undefined, `argpi((0,0))`
...
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemenal function.
+
+`deg2rad` converts phase angles from degrees to radians.
```
Comment (jvdp1) [inline doc/specs/stdlib_math.md:395]:
```suggestion
### Description

`deg2rad` converts phase angles from degrees to radians.
```

File: doc/specs/stdlib_math.md [hunk 2]
```diff
@@ -382,6 +382,66 @@ Notes: Although the angle of the complex number `0` is undefined, `argpi((0,0))`
...
+
+`result = ` [[stdlib_math(module):deg2rad(interface)]] `(theta)`
+
+#### Arguments
+
+`theta`: Shall be a `real` scalar/array.
+
+#### Return value
+
+Returns the `real` phase angle in radians
```
Comment (jvdp1) [inline doc/specs/stdlib_math.md:407]:
```suggestion
Returns the `real` phase angle in radians.
```

File: doc/specs/stdlib_math.md [hunk 3]
```diff
@@ -382,6 +382,66 @@ Notes: Although the angle of the complex number `0` is undefined, `argpi((0,0))`
...
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemenal function.
+
+`rad2deg` converts phase angles from radians to degrees.
```
Comment (jvdp1) [inline doc/specs/stdlib_math.md:425]:
```suggestion
### Description

`rad2deg` converts phase angles from radians to degrees.
```

File: src/stdlib_math.fypp [hunk 4]
```diff
@@ -453,6 +473,22 @@ contains
     end function argpi_${k1}$
     #:endfor
 
+    #:for k1, t1 in REAL_KINDS_TYPES
+    elemental function deg2rad_${k1}$(theta)result(result)
```
Comment (jvdp1) [inline src/stdlib_math.fypp:477]:
```suggestion
    elemental function deg2rad_${k1}$(theta) result(result)
```

File: src/stdlib_math.fypp [hunk 5]
```diff
@@ -453,6 +473,22 @@ contains
...
+        ${t1}$, intent(in) :: theta
+        ${t1}$ :: result
+        result = theta * PI_${k1}$ / 180.0_${k1}$
+        
+    end function deg2rad_${k1}$
+    
+    elemental function rad2deg_${k1}$(theta)result(result)
```
Comment (jvdp1) [inline src/stdlib_math.fypp:484]:
```suggestion
    elemental function rad2deg_${k1}$(theta) result(result)
```

**Review by jvdp1 — APPROVED** [review]
LGTM. Thank you @MilanSkocic . Pending minor comments, changes, this PR can be merged IMO.

**Review by jvdp1 — APPROVED** [review]
Thank you @MilanSkocic . IMO it is good to be merged. Let wait a couple of days for other potential comments/reviews.

**Comment by jalvesz** [comment]
LGTM @MilanSkocic thanks for this nice addition!!

**Comment by jvdp1** [comment]
Thank you @MilanSkocic for this addition. I will merge this PR.

**Pull request merged**